include/lsst/meas/astrom/SipTransform.h

```diff
@@ -0,0 +1,357 @@
+ */
+class SipReverseTransform : public SipTransformBase {
+public:
+
+    /**
+     *  Convert a PolynomialTransform to an equivalent SipReverseTransform.
+     *
+     *  @param[in]   poly           PolynomialTransform to convert.
+     *  @param[in]   pixelOrigin    CRPIX @f$(u_0,v_0)@f$
+     *  @param[in]   cdMatrix       CD matrix @f$Z@f$
+     */
+    static SipReverseTransform convert(
+        PolynomialTransform const & poly,
+        afw::geom::Point2D const & pixelOrigin,
+        afw::geom::LinearTransform const & cdMatrix
+    );
+
+    /**
+     *  Convert a ScaledPolynomialTransform to an equivalent SipReverseTransform.
+     *
+     *  @param[in]   scaled         ScaledPolynomialTransform to convert.
+     *  @param[in]   pixelOrigin    CRPIX @f$(u_0,v_0)@f$
+     *  @param[in]   cdMatrix       CD matrix @f$Z@f$
+     */
+    static SipReverseTransform convert(
+        ScaledPolynomialTransform const & scaled,
+        afw::geom::Point2D const & pixelOrigin,
+        afw::geom::LinearTransform const & cdMatrix
+    );
+
+    /**
+     *  Convert a ScaledPolynomialTransform to an equivalent SipReverseTransform.
+     *
+     *  The pixel origin CRPIX and CD matrix are defined to reproduce the translation
+     *  and linear transformation in the ScaledPolynomialTransforms output and
+     *  input scalings (respectively).
+     */
+    static SipReverseTransform convert(ScaledPolynomialTransform const & scaled);
+
+    /**
+     *  Construct a SipReverseTransform from its components.
+     *
+     *  @param[in]   pixelOrigin    CRPIX @f$(u_0,v_0)@f$
+     *  @param[in]   cdMatrix       CD matrix @f$Z@f$
+     *  @param[in]   reverseSipPoly Polynomial transform @f$(AP,BP)@f$
+     */
+    SipReverseTransform(
+        afw::geom::Point2D const & pixelOrigin,
+        afw::geom::LinearTransform const & cdMatrix,
+        PolynomialTransform const & reverseSipPoly
+    ) : SipTransformBase(pixelOrigin, cdMatrix, reverseSipPoly),
+        _cdInverse(cdMatrix.invert())
+    {}
+
+    SipReverseTransform(SipReverseTransform const & other) = default;
+
+    SipReverseTransform(SipReverseTransform && other) = default;
+
+    SipReverseTransform & operator=(SipReverseTransform const & other) = default;
+
+    SipReverseTransform & operator=(SipReverseTransform && other) = default;
+
+    void swap(SipReverseTransform & other) {
+        SipTransformBase::swap(other);
+        std::swap(_cdInverse, other._cdInverse);
+    }
+
+    /**
+     * Return an approximate affine transform at the given point.
+     */
+    afw::geom::AffineTransform linearize(afw::geom::Point2D const & in) const;
+
+    /**
+     * Apply the transform to a point.
+     */
+    afw::geom::Point2D operator()(afw::geom::Point2D const & xy) const;
+
+private:
+    friend class PolynomialTransform;
+    friend class ScaledPolynomialTransform;
+    afw::geom::LinearTransform _cdInverse;
+};
+
+/**
+ *  Create a new TanWcs from a pair of SIP transforms and the sky origin.
```

Comment:
I suggest you say "TAN SIP WCS" instead of a "TanWcs"; the latter is a poorly named class that should go away; the former is a more general concept that should remain after the WC rewrite.